modules/gapi/include/opencv2/gapi/cpu/gcpukernel.hpp

```diff
@@ -287,6 +315,38 @@ class GCPUKernelImpl: public cv::detail::OCVCallHelper<Impl, typename K::InArgs,
 
 #define GAPI_OCV_KERNEL(Name, API) struct Name: public cv::GCPUKernelImpl<Name, API>
 
+class gapi::cpu::GOCVFunctor : public gapi::GFunctor
+{
+public:
+    using Impl = std::function<void(GCPUContext &)>;
+
+    GOCVFunctor(const char* id, const Impl& impl)
+        : gapi::GFunctor(id), impl_{GCPUKernel(impl)}
+    {
+    }
+
+    GKernelImpl impl()       const override { return impl_;                    }
+    gapi::GBackend backend() const override { return gapi::cpu::backend(); }
+
+private:
+    GKernelImpl impl_;
+};
+
+
+template<typename K, typename Callable>
+gapi::cpu::GOCVFunctor gapi::cpu::ocv_kernel(Callable& c)
+{
+    using P = detail::OCVCallHelper<Callable, typename K::InArgs, typename K::OutArgs>;
+    return GOCVFunctor(K::id(), std::bind(&P::callFunctor, std::placeholders::_1, std::ref(c)));
+}
+
+template<typename K, typename Callable>
+gapi::cpu::GOCVFunctor gapi::cpu::ocv_kernel(const Callable& c)
+{
+    using P = detail::OCVCallHelper<Callable, typename K::InArgs, typename K::OutArgs>;
+    return GOCVFunctor(K::id(), std::bind(&P::callFunctor, std::placeholders::_1, c));
+}
```

Comment:
What's the reason to have two overloads for ocv_kernel here?

Reply:
`Callable&` for functors with state we have to avoid copying
`const Callable&` for lambdas

Different in `std::bind` function, for functors using `std::ref` to avoid copying, for lambdas not.
Will be great if we can using universal ref `Callable&&` here, but I don't know how to use perfect forwarding here